vello_tests/src/lib.rs

```diff
@@ -171,6 +176,20 @@ pub fn write_png_to_file(
     let mut writer = encoder.write_header()?;
     writer.write_image_data(image.data.data())?;
     writer.finish()?;
+    let size = file.metadata().unwrap().len();
+    drop(file);
+    let oversized_path = out_path.with_extension("oversized.png");
+    if max_size_in_bytes.is_some_and(|max_size_in_bytes| size > max_size_in_bytes) {
+        std::fs::rename(out_path, &oversized_path)?;
+        bail!(
+            "File was oversized, expected {} bytes, got {size} bytes. New file written to {to}",
+            max_size_in_bytes.unwrap(),
+            to = oversized_path.display()
+        );
+    } else {
+        // Intentionally do not handle errors here
+        drop(std::fs::remove_file(oversized_path));
```

Comment:
Just for my understanding, why is this removed? It shouldn't even be there should it? Or is this because when a previous image resulted `oversized` and is now not anymore, that this will be cleaned up?

Reply:
Yeah, if the previous run was oversized, we no longer need this file.

It's pretty poor UX to have old snapshot files "hanging around"